cpg_workflows/large_cohort/sample_qc.py

```diff
@@ -82,6 +83,86 @@ def initialise_sample_table() -> hl.Table:
     return ht
 
 
+def generate_sex_coverage_mt(
+    vds: hl.vds.VariantDataset,
+    calling_intervals: hl.Table,
+    tmp_prefix: Path,
+):
+    """
+    generate the pre-computed MT of coverage intervals, to supply to annotate_sex
+    Args:
+        vds ():
+        calling_intervals ():
+        tmp_prefix ():
+
+    Returns:
+
+    """
+    key_dtype = calling_intervals.key.dtype
+    if (
+            len(key_dtype) != 1 or
+            not isinstance(calling_intervals.key[0].dtype, hl.tinterval) or
+            calling_intervals.key[0].dtype.point_type != vds.reference_data.locus.dtype
+    ):
+        raise ValueError(f'"impute_sex_chromosome_ploidy": expect calling_intervals to be list of intervals or '
+                         f'table with single key of type interval<locus>, found table with key: {key_dtype}')
+
+    rg = vds.reference_data.locus.dtype.reference_genome
+    par_boundaries = []
+    for par_interval in rg.par:
+        par_boundaries.append(par_interval.start)
+        par_boundaries.append(par_interval.end)
+
+    # segment on PAR interval boundaries
+    calling_intervals = hl.segment_intervals(calling_intervals, par_boundaries)
```

Comment:
I think PAR regions are already excluded from the calling_intervals ht?

Reply:
If you're confident, then we can remove that section. If there's any chance PAR. regions remain this should be a free operation. I was just copy pasting 😅